bindings/python/examples/iio_attr.py

```diff
@@ -0,0 +1,480 @@
+        if self.arguments.device is not None and \
+                not _str_match(dev.name, self.arguments.device, self.arguments.ignore_case):
+            return
+
+        if dev is None:
+            print("\t" + dev.id + ":", end="")
+
+            if dev.name is not None:
+                print(" " + dev.name, end="")
+
+            print(", ", end="")
+
+            if self.arguments.search_channel and self.arguments.device is None:
+                print("found " + str(len(dev.channels)) + " channels")
+
+        for channel in dev.channels:
+            self._channel_information(dev, channel)
+
+    def _channel_information(self, dev, channel):
+        channel_stop = not self.arguments.search_channel or self.arguments.device is None
+        input_stop = self.arguments.input_only and channel.output
+        output_stop = self.arguments.output_only and not channel.output
+        scan_stop = self.arguments.scan_only and not channel.scan_element
+
+        if channel_stop or input_stop or output_stop or scan_stop:
+            return
+
+        type_name = "output" if channel.output else "input"
+
+        if not _str_match(self.arguments.device, dev.name, self.arguments.ignore_case):
+            return
+
+        if self.arguments.channel is not None and \
+                not _str_match(channel.id, self.arguments.channel, self.arguments.ignore_case) and \
+                (channel.name is None or (channel.name is not None and
+                                          not _str_match(channel.name, self.arguments.channel,
+                                                         self.arguments.ignore_case))):
+            return
+
+        if (not self.arguments.scan_only and self.arguments.channel is None) or \
+                (self.arguments.scan_only and channel.scan_element):
+            print("dev \'" + dev.name + "\', channel \'" + channel.id + "\'", end="")
+
+            if channel.name is not None:
+                print(", id \'" + channel.name + "\'", end="")
+
+            print(" (" + type_name, end="")
+
+            if channel.scan_element:
+                self._scan_channel_information(channel)
+            else:
+                print("), ", end="")
+
+        self._channel_attributes_information(dev, channel)
+
+    def _scan_channel_information(self, channel):
+        sign = "s" if channel.data_format.is_signed else "u"
+        if channel.data_format.is_fully_defined:
+            sign = sign.upper()
+
+        if channel.data_format.repeat > 1:
+            print("X" + str(channel.data_format.repeat), end="")
+
+        print(", index: " + str(channel.index) + ", format: "
+              + "b" if channel.data_format.is_be else "l"
+              + "e:" + sign + str(channel.data_format.bits)
+              + "/" + str(channel.data_format.length) + str(channel.data_format.repeat)
+              + ">>" + str(channel.data_format.shift))
+
+        print() if self.arguments.scan_only else print(", ", end="")
```

Comment:
To fix pep8 try this: `print("" if self.arguments.scan_only else ", ", end="")`

Reply:
Solved.